Source/ChilliSource/Core/Base/Application.h

```diff
@@ -311,6 +311,18 @@ namespace ChilliSource
         /// @return A const pointer to the widget factory system.
         ///
         const WidgetFactory* GetWidgetFactory() const noexcept;
+        
+        /// Takes a snapshot of the given scene and renders it to its
+        /// render target during the render stage of the application.
+        /// This is used mostly for single shot render to texture
+        /// NOTE: This means the texture of the target will not be populated til next frame.
+        ///
+        /// @param scene
+        ///     Scene that should be rendered
+        /// @param target
+        ///     Optional, if wanting to render the scene to a target other than its own
+        ///
+        void RenderScene(Scene* scene, TargetGroup* target = nullptr) noexcept;
```

Comment:
I did originally add it to scene but the coupling with Application was really tight. However I never quite liked this being part of application. I'll move it back to scene if the coupling is not an issue